highfidelity/hifi
# Avatar Exporter - reference thumb rotations

Addresses:
https://highfidelity.manuscript.com/f/cases/20488

Also minor fix to enforce fixed text field height in the Export New Avatar window, as well as tweaks to the README file including recommended Unity version and link to full documentation.

Test Plan:

Note: Post-merge testing is not necessary as this PR will not affect the Interface build after merging.

Note: Test must be run on PC.

- Download the following files:
https://github.com/highfidelity/hifi/raw/0e89e37c8edb94ca2131cfd71c2914da13278b23/tools/unity-avatar-exporter/avatarExporter.unitypackage
https://hifi-content.s3.amazonaws.com/davidback/avatars/zonneko.fbx
https://hifi-content.s3.amazonaws.com/davidback/avatars/Mason_Idle.fbx
- Ensure you have Unity version between 2017.4.17f1 and 2018.2.12f1 installed, and open a new Unity project
- Open the above avatarExporter.unitypackage and import it into your Unity project
- Open the README file in your Assets window and verify it includes updated instructions such as recommended Unity version at the top and a link to the documentation at the bottom
- Follow the README instructions to export the above 2 avatars using the Export New Avatar option
- Within the Export New Avatar window, verify when typing text into the Project Name field and deleting it that the UI within the window does not shift
- Upload resulting .fst and .fbx if you have access to an S3 bucket, or ask me to upload them for you
- Verify on any latest dev build that the avatar for the uploaded .fst appears correctly and animates correctly including with the Emote app, and specifically examine the thumbs and ensure they're oriented correctly including during animations
